pkg/gds/config/config_test.go

```diff
@@ -103,43 +105,54 @@ func TestConfig(t *testing.T) {
 }
 
 func TestRequiredConfig(t *testing.T) {
-	// Set required environment variables and cleanup after
-	prevEnv := curEnv(
+	required := []string{
 		"GDS_DATABASE_URL",
 		"GDS_SECRET_KEY",
-		"GDS_REPLICA_ENABLED",
 		"GDS_REPLICA_PID",
 		"GDS_REPLICA_REGION",
-	)
-	t.Cleanup(func() {
+		"GDS_ADMIN_TOKEN_KEYS",
+	}
+
+	// Collect required environment variables and cleanup after
+	prevEnv := curEnv(required...)
+	cleanup := func() {
 		for key, val := range prevEnv {
 			if val != "" {
 				os.Setenv(key, val)
 			} else {
 				os.Unsetenv(key)
 			}
 		}
-	})
-
-	_, err := config.New()
-	require.Error(t, err)
-	setEnv(
-		"GDS_DATABASE_URL",
-		"GDS_SECRET_KEY",
-		"GDS_REPLICA_ENABLED",
-		"GDS_REPLICA_PID",
-		"GDS_REPLICA_REGION",
-	)
+	}
+	t.Cleanup(cleanup)
 
+	// Ensure that we've captured the complete set of required environment variables
+	setEnv(required...)
 	conf, err := config.New()
 	require.NoError(t, err)
 
+	// Ensure that each environment variable is required
+	for _, envvar := range required {
+		// Add all environment variables but the current one
+		for _, key := range required {
+			if key == envvar {
+				os.Unsetenv(key)
+			} else {
+				setEnv(key)
+			}
+		}
+
+		_, err := config.New()
+		require.Errorf(t, err, "expected %q to be required but no error occurred", envvar)
+	}
```

Comment:
Changed the test so that it sets all environment variables to test that all the required variables are in fact required (the previous test only required one of the required environment variables to be required to pass).